app/models/piece.rb

```diff
@@ -93,9 +93,24 @@ def on_board?(end_point)
 
 	def do_move!(pos_x, pos_y)
 		self.update_attributes(pos_x: pos_x, pos_y: pos_y)
+  end
+
+  def is_occupied?(x, y)
+    game.pieces.where(pos_x: x, pos_y: y, active: true).present?
+  end
 
- 
-   	
+  def move_to!(dest_x, dest_y)
+    @game = game
```

Comment:
do not use instance variables unless you have a good reason to do so. this method should replace 
@game
@dest_occupied
@captured 

with

game
dest_occupied
captured